examples/cartesian_reconstruction.py

```diff
@@ -0,0 +1,98 @@
+"""
+Neuroimaging cartesian reconstruction
+=====================================
+
+Author: Chaithya G R
+
+In this tutorial we will reconstruct an MRI image from the sparse kspace
+measurments.
+
+Import neuroimaging data
+------------------------
+
+We use the toy datasets available in pysap, more specifically a 2D brain slice
+and the acquisition cartesian scheme.
+"""
+
+# Package import
+from modopt.math.metrics import ssim
+from mri.numerics.reconstruct import sparse_rec_fista
+from mri.numerics.fourier import FFT2
+from mri.numerics.utils import generate_operators
+from mri.numerics.utils import convert_mask_to_locations
+import pysap
+
+# Third party import
+import numpy as np
+
+# Loading input data
+image = pysap.utils.load_image('../../../Data/Pysap_examples/base_image.npy')
+
+# Obtain K-Space Cartesian Mask
+mask = pysap.utils.load_image(
+    '../../../Data/Pysap_examples/mask_cartesian.npy')
+
+# View Input
+image.show()
+mask.show()
+
+#############################################################################
+# Generate the kspace
+# -------------------
+#
+# From the 2D brain slice and the acquisition mask, we generate the acquisition
+# measurements, the observed kspace.
+# We then reconstruct the zero order solution.
+
+
+# Get the locations of the kspace samples
+kspace_loc = convert_mask_to_locations(np.fft.fftshift(mask.data))
+# Generate the subsampled kspace
+fourier_op = FFT2(samples=kspace_loc, shape=image.shape)
+kspace_data = fourier_op.op(image)
+
+# Zero order solution
+image_rec0 = pysap.Image(data=fourier_op.adj_op(kspace_data),
+                         metadata=image.metadata)
+image_rec0.show()
+
+# Calculate SSIM
+base_ssim = ssim(image_rec0, image)
+print(base_ssim)
+
+#############################################################################
+# FISTA optimization
+# ------------------
+#
+# We now want to refine the zero order solution using a FISTA optimization.
+# Here no cost function is set, and the optimization will reach the
+# maximum number of iterations.
+
+# Generate operators
+gradient_op, linear_op, prox_op, cost_op = generate_operators(
+    data=kspace_data,
+    wavelet_name="sym8",
+    samples=kspace_loc,
+    nb_scales=4,
+    mu=2.364e-07,
+    non_cartesian=False,
+    uniform_data_shape=None,
+    gradient_space="synthesis",
+    padding_mode="periodization")
+
+# Start the FISTA reconstruction
+max_iter = 200
```

Comment:
have you checked that we are at convergence?

Reply:
Nope. Will check. I am afraid I need to see how to set up the Modopt for correct convergence checks to prevent manual checks. 
But I do see cost function flattening out at tail in plot

Reply:
Ok then it's fine, for now manual checking will do